app/services/pipeline_report_service.rb

```diff
@@ -146,46 +160,32 @@ def fetch_taxon_counts(_pipeline_run_id, _background_id)
     return taxon_counts_and_summaries_query.pluck(*FIELDS_TO_PLUCK)
   end
 
-  def zero_metrics(tax_id, tax_level, count_type, mean, stdev)
-    # Fill in default zero values for FIELDS_TO_PLUCK.
+  def zero_metrics(taxon)
+    # Fill in default zero values if a taxon is missing fields.
     # Necessary for taxons absent from sample to match the taxon_counts_and_summaries structure,
     # since they're fetched from TaxonSummary, which doesn't have some columns listed in FIELDS_TO_PLUCK.
-    # The two nil values (genus_taxid and name) will be filled in when counts_by_tax_level is
-    # transformed to be indexed by taxid and count type.
-    [
-      tax_id,
-      nil,
-      count_type,
-      tax_level,
-      0,
-      0,
-      0,
-      0,
-      mean,
-      stdev,
-      nil,
-    ]
+    FIELDS_INDEX.each do |field, index|
+      taxon[index] = FIELDS_DEFAULTS[field] unless taxon[index]
+    end
+    return taxon
   end
 
   def fetch_taxons_absent_from_sample(_pipeline_run_id, _background_id)
+    tax_ids = TaxonCount.select(:tax_id).where(pipeline_run_id: @pipeline_run_id).distinct.pluck(:tax_id)
```

Comment:
Just remove the pluck here.
You should not pluck here. You can inject he query directly and then rails is smart enough to not do 2 queries.